src/libsync/propagateremotemove.cpp

```diff
@@ -85,6 +85,31 @@ void PropagateRemoteMove::start()
 
     if (origin == _item->_renameTarget) {
         // The parent has been renamed already so there is nothing more to do.
+
+        if (!_item->_encryptedFileName.isEmpty()) {
+            const auto path = _item->_file;
+            const auto slashPosition = path.lastIndexOf('/');
+            const auto parentPath = slashPosition >= 0 ? path.left(slashPosition) : QString();
+
+            SyncJournalFileRecord parentRec;
+            bool ok = propagator()->_journal->getFileRecord(parentPath, &parentRec);
+            if (!ok) {
+                done(SyncFileItem::NormalError);
+                return;
+            }
+
+            // We should be encrypted as well since our parent is
+            const auto remoteParentPath = parentRec._e2eMangledName.isEmpty() ? parentPath : parentRec._e2eMangledName;
+
+            const auto lastSlashPosition = _item->_encryptedFileName.lastIndexOf('/');
+            const auto encryptedName = slashPosition >= 0 ? _item->_encryptedFileName.mid(lastSlashPosition + 1) : QString();
+
+            if (!encryptedName.isEmpty()) {
+                _item->_encryptedFileName = remoteParentPath + "/" + encryptedName;
```

Comment:
I think this warrants a comment on why this work, because I guess this works only if the move is due to a non encrypted parent which got renamed right? (because otherwise we'd do a remove + new)

I'm just concerned we wouldn't remember that subtlety 6 months from now. I'm not even sure I got it right now. :-D

Reply:
@er-vin I'll explain this during the call on Monday :)

Reply:
Fair enough. :-)

But that kind of prove my point it needs a comment. OK, also proves I'm likely too stupid for that code. :-D

Reply:
@er-vin Nah. It's just, this place needs to be changed and the changes are affecting the other part that happens later :)